### PR TITLE
JBPM-7537: [GSS] (7.0.z) Cannot use multibyte class name for Data Objects

### PR DESCRIPTION
**Pull request by wmedvede**

### CODE REVIEW

**Comment by wmedvede** [comment]
Hello @pefernan @jsoltes would you mind review this PR?
Thanks

**Comment by wmedvede** [comment]
Jenkins execute full downstream build

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/test/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerServiceHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        //test when the package already exists
+        Package existingPackage = mock(Package.class);
+        when(ioService.exists(any(org.uberfire.java.nio.file.Path.class))).thenReturn(true);
+        when(moduleService.resolvePackage(any(Path.class))).thenReturn(existingPackage);
+        result = serviceHelper.ensurePackageStructure(module, newPackageName);
+        assertEquals(existingPackage, result);
+        verify(moduleService, never()).newPackage(any(Package.class), anyString());
+
+        //test when the package not exists
+        Package createdPackage = mock(Package.class);
```
Comment (jsoltes) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/test/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerServiceHelperTest.java]: This code could be refactored using some helping method (as it is basically the same as above). Also my first thought seeing the comments starting //test... - this test could be split into three separate tests.

**Comment by wmedvede** [comment]
@jsoltes test was splited and improved.  Mind take a look so we can move this forward?
Thanks!

**Comment by wmedvede** [comment]
Jenkins execute full downstream build

**Comment by wmedvede** [comment]
not sure what's going on here, rebased with master and pushed an hour ago or so, but nothing indicates that a build is being processed for this PR. Let's give it a manual try

**Comment by wmedvede** [comment]
Jenkins execute full downstream build

**Comment by wmedvede** [comment]
Hello @manstis 
This was rebased, etc, but continues complaining due to variants of org.kie.workbench.common.services.backend.compiler.kie.KieDefaultMavenCompilerTest.XXXXs

This time the single-repository build failed, the full build failed, but at least the https://kie-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/kie-wb-common-downstream-pullrequests/1349/artifact/downstream-repos/kie-wb-distributions/kie-wb-parent/kie-wb-distribution-wars/kie-wb/target/kie-wb-7.10.0-SNAPSHOT-wildfly11.war was generated properly.

Starting to have bad feelings about the time/attempts for getting a green :(.
Since I'm leaving for PTO tomorrow and won't be here for the 7.1 cut date I strongly suggest to merge this, so I can double check after merging, etc. In other case with high probability nobody else will be watching, rebasing, and re-re-re launching the build on this PR. But we need this in for the 7.1.

Thanks

**Comment by manstis** [comment]
@porcelli @desmax74 Wasn't there a PR merged recently to disable the daemons for your new backend tests? If this continues to fail (as I have seen other ```kie-wb-common``` PRs fail) for the same reasons you may have missed some tests!?!

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
@wmedvede Let's see what happens with the last build request I made... IIRC @porcelli and @desmax74 PR was merged yesterday... please let me know the outcome (I have bookmarked this PR to check later; but if you see a result first please let me know).

**Comment by wmedvede** [comment]
@manstis Thanks.

for the record: the commit below was included in my PR last time it was rebased and re-pushed Yesterday.

commit 7ef0bb2555ec681ca998ec27ec6dabef6e8dd906
Author: Massimiliano Dessì <desmax74@yahoo.it>
Date:   Wed Aug 8 12:30:35 2018 +0200

    Improve test stability by disabling git daemons

**Pull request merged**